contracts/utils/UniversalResolver.sol

```diff
@@ -42,45 +77,124 @@ contract UniversalResolver is IExtendedResolver, ERC165 {
     function resolve(bytes calldata name, bytes memory data)
         external
         view
-        override
         returns (bytes memory, address)
     {
+        return
+            _resolveSingle(
+                name,
+                data,
+                batchGatewayURLs,
+                this.resolveSingleCallback.selector,
+                ""
+            );
+    }
+
+    function resolve(bytes calldata name, bytes[] memory data)
+        external
+        view
+        returns (bytes[] memory, address)
+    {
+        return resolve(name, data, batchGatewayURLs);
+    }
+
+    function resolve(
+        bytes calldata name,
+        bytes memory data,
+        string[] memory gateways
+    ) external view returns (bytes memory, address) {
+        return
+            _resolveSingle(
+                name,
+                data,
+                gateways,
+                this.resolveSingleCallback.selector,
+                ""
+            );
+    }
+
+    function resolve(
+        bytes calldata name,
+        bytes[] memory data,
+        string[] memory gateways
+    ) public view returns (bytes[] memory, address) {
+        return
+            _resolve(name, data, gateways, this.resolveCallback.selector, "");
+    }
+
+    function _resolveSingle(
+        bytes calldata name,
+        bytes memory data,
+        string[] memory gateways,
+        bytes4 callbackFunction,
+        bytes memory metaData
```

Comment:
`metaData` is used in `reverseCallback` so that the resolved name and reverse resolver address can be stored while forward resolution is done

Reply:
`_resolveSingle` is used by `getForwardDataFromReverse`, which passes through metadata so that the resolvedName and reverseResolverAddress can be restored in the callback